ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/healthcheck/HealthCheckAgent.java

```diff
@@ -58,22 +59,23 @@
 
   private final SimpleKsqlClient ksqlClient;
   private final URI serverEndpoint;
-  private final ServiceContext serviceContext;
   private final KsqlConfig ksqlConfig;
   private final CommandRunner commandRunner;
+  private final Admin adminClient;
 
   public HealthCheckAgent(
       final SimpleKsqlClient ksqlClient,
       final KsqlRestConfig restConfig,
       final ServiceContext serviceContext,
       final KsqlConfig ksqlConfig,
-      final CommandRunner commandRunner
+      final CommandRunner commandRunner,
+      final Admin adminClient
```

Comment:
Are we making a `HealthCheckAgent` for the internal and external tenents?

Reply:
Hmm that is an interesting point. The current implementation of `HealthCheckAgent` just checks to see what kind of permissions we have on the command topic and then assumes health based on that, but maybe we should also check to see if we can describe the processing log, since they're on different tenants

Reply:
The intent behind this check is to assess whether ksql can communicate with its backing Kafka for the purpose of serving the ksqlDB API. So IMO checking that we can access the internal tenant here is ok.

As discussed, we should file a follow-up ticket to remove the checks for authorization and not-found exceptions once we've finished migrating every cluster to internal tenants.

Reply:
Filed a follow-up ticket